src/hazardous/kdf/mod.rs

```diff
@@ -26,6 +26,10 @@ pub mod hkdf;
 /// PBKDF2(Password-Based Key Derivation Function 2) as specified in the [RFC 8018](https://tools.ietf.org/html/rfc8018).
 pub mod pbkdf2;
 
+#[cfg(feature = "alloc")]
```

Comment:
IIUC, this should work on both `std` and `alloc`-specific features (correct me if I'm wrong). I think we can make the conditionals just like those for `argon2i`.

Also with the added `cfg_attr` which highlights these features nicely in rendered docs. 

```suggestion
#[cfg(any(feature = "safe_api", feature = "alloc"))]
#[cfg_attr(docsrs, doc(cfg(any(feature = "safe_api", feature = "alloc"))))]
```

Reply:
Thanks. Added.